src/euler/EulerWrapper.sol

```diff
@@ -0,0 +1,70 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.30;
+
+import { BaseWrapper } from "../BaseWrapper.sol";
+import { IEFlashLoanCallback } from "./interfaces/IEFlashLoanCallback.sol";
+import { IEVKFactoryPerspective } from "./interfaces/IEVKFactoryPerspective.sol";
+import { IEVault } from "./interfaces/IEVault.sol";
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+/// @dev Euler Flash Lender that uses Euler as source of liquidity.
+contract EulerWrapper is BaseWrapper, IEFlashLoanCallback {
+    error UnsupportedAsset(address asset);
+    error UnverifiedVault(address vault);
+
+    event VaultsAdded(uint256 count);
+
+    mapping(IERC20 token => IEVault vault) public vaults;
```

Comment:
you can have multiple vaults for the same asset, I'd make it an array so then you can find a vault that has enough liquidity, and prob sort the array every time we add a new vault for an asset, maybe we can use the IterableSet (or however is called) from OZ so we already remove duplicates? (albeit that'd be unordered, but not sure how much we care about the order as the set is not likely to be big(

Reply:
Yes, thanks, I didn't think about multiple vaults per token. I've rewritten that logic. I think we can omit a complex data structure to sort vaults by liquidity. Just use an array and iterate through it.